parsl/app/bash.py

```diff
@@ -65,7 +65,7 @@ def open_std_fd(fdname):
 
     returncode = None
     try:
-        proc = subprocess.Popen(executable, stdout=std_out, stderr=std_err, shell=True, executable='/bin/bash')
+        proc = subprocess.Popen(executable, stdout=std_out, stderr=std_err, shell=True, executable='/bin/bash', close_fds=False)
```

Comment:
Yeah, it will affect all executors in theory. In practice, though, do they check all their open fds, or do something else that would be affected by this change? 

The problem is that Flux provides MPI to jobs by letting them inherit a FD and setting an environment variable defining what the FD is. So in `MPI_Init` the MPI implementation will read the environment variable and go to open the FD, which fails if `close_fds=True`. 

There may be other ways to provide MPI, though, so the problem is specific to that specific method of providing it, and to executors that launch MPI jobs.

Reply:
I think its fine - was interested in having the reasoning recorded here in case something interesting shows up later.